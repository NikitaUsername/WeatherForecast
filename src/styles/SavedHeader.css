.savedHeaderTitle {
    margin-top: 1rem;
}

.addCityForm{
    margin-top: 1em;
}

.addCityForm__inputFld {
    background-color: #dddddd;
    height: fit-content;
    border-radius: 0.65rem;
}
.addCityForm__addButton{
    margin-top: 0;
}