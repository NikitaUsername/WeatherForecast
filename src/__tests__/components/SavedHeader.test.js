import SavedHeader from '../../components/SavedHeader'
import React from 'react'
import renderer from 'react-test-renderer'
import { Provider } from 'react-redux'
import configureMockStore from 'redux-mock-store'
import thunk from 'redux-thunk'

const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);



describe('SavedHeader component', () => {
    test('matches the snapshot withought cities', () => {

        const store = mockStore({
            cities: [
            ]
        });

        const tree = renderer
            .create(
                <Provider store={store}>
                    <SavedHeader />
                </Provider>).toJSON();
        expect(tree).toMatchSnapshot();
    });

    test('matches the snapshot with loader', () => {

        const store = mockStore({
            cities: [
                {
                    name: 'Moscow',
                    weather: {}
                } 
            ]
        });

        const tree = renderer
            .create(
                <Provider store={store}>
                    <SavedHeader />
                </Provider>).toJSON();
        expect(tree).toMatchSnapshot();
    });

    test('matches the snapshot with city data', () => {

        const store = mockStore({
            cities: [
                {
                    name: 'Irkutsk',
                    weather: {
                
                        weather: [
                            { id: 500, main: "Rain", description: "light rain", icon: "10n" }
                        ],
                        main: {
                            temp: 20.46,
                            feels_like: 19.46,
                            temp_min: 19.44,
                            temp_max: 21.67,
                            pressure: 1021,
                            humidity: 83
                        },
                        wind: {
                            speed: 5.1,
                            deg: 40
                        },
                        clouds: {
                            all:90
                        }
                    }
                }
            ]
        });

        const tree = renderer
            .create(
                <Provider store={store}>
                    <SavedHeader />
                </Provider>).toJSON();
        expect(tree).toMatchSnapshot();
    });
})