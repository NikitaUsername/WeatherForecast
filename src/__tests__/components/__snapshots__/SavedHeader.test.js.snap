// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SavedHeader component matches the snapshot with city data 1`] = `
<div
  className="container"
>
  <div
    className="row"
  >
    <div
      className="col-lg-4"
    >
      <h2
        id="title"
      >
        Избранное
      </h2>
    </div>
    <div
      className="col-lg-6"
    >
      <form
        onSubmit={[Function]}
      >
        <input
          id="inputfld"
          placeholder="Город"
        />
        <input
          id="update-button"
          type="submit"
          value="Добавить в избранное"
        />
      </form>
    </div>
  </div>
  <div
    className="row"
  >
    <div
      className="container"
    >
      <div
        className="row"
      >
        <div
          className="col-lg-4"
        >
          <h2
            id="cityname"
          >
            Irkutsk
          </h2>
        </div>
        <div
          className="col-lg-6"
        >
          <div
            className="row"
            
          >
            <p>
              Погода
            </p>
            <div
              id="minival"
            >
              light rain
            </div>
          </div>
          <div
            className="row"
            
          >
            <p>
              Давление
            </p>
            <div
              id="minival"
            >
              1021
               hPa
            </div>
          </div>
          <div
            className="row"
            
          >
            <p>
              Скорость ветра
            </p>
            <div
              id="minival"
            >
               
              5.1
               m/s
            </div>
          </div>
          <div
            className="row"
            
          >
            <p>
              Облачность
            </p>
            <div
              id="minival"
            >
               
              90
               %
            </div>
          </div>
        </div>
        <div
          className="col-lg-2"
        >
          <button
            className="close"
            onClick={[Function]}
          >
            <span
              aria-hidden="true"
            >
              ×
            </span>
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`SavedHeader component matches the snapshot with loader 1`] = `
<div
  className="container"
>
  <div
    className="row"
  >
    <div
      className="col-lg-4"
    >
      <h2
        id="title"
      >
        Избранное
      </h2>
    </div>
    <div
      className="col-lg-6"
    >
      <form
        onSubmit={[Function]}
      >
        <input
          id="inputfld"
          placeholder="Город"
        />
        <input
          id="update-button"
          type="submit"
          value="Добавить в избранное"
        />
      </form>
    </div>
  </div>
  <div
    className="row"
  >
    <div
      className="container"
    >
      <div
        className="row"
      >
        <div
          className="col-lg-12"
        >
          <h2
            id="waiting"
          >
            Подождите, данные загружаются
          </h2>
        </div>
        <div
          className="col-lg-12 "
        >
          <div
            className="center-block"
          >
            <div
              className="css-236bqq"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`SavedHeader component matches the snapshot withought cities 1`] = `
<div
  className="container"
>
  <div
    className="row"
  >
    <div
      className="col-lg-4"
    >
      <h2
        id="title"
      >
        Избранное
      </h2>
    </div>
    <div
      className="col-lg-6"
    >
      <form
        onSubmit={[Function]}
      >
        <input
          id="inputfld"
          placeholder="Город"
        />
        <input
          id="update-button"
          type="submit"
          value="Добавить в избранное"
        />
      </form>
    </div>
  </div>
  <div
    className="row"
  />
</div>
`;
